Beam: handle commas in Sentry DSN parameter (close #391)

benjben opened the pull request:
Logic is:
- if `--sentry-dsn=foo` then `foo` is put in `List(foo)`
- if `--sentry-dsn=a, b,c` then `a` `b` and `c` are put in `List(a, b,c)`
So by calling `args.asMap.get("sentry-dsn").map(_.mkString(","))` we are sure to reconstruct it back and we don't need base64 encoding/decoding

chuwy commented on modules/beam/src/main/scala/com.snowplowanalytics.snowplow.enrich.beam/config.scala:
```diff
@@ -78,7 +78,7 @@ object config {
         args.optional("enrichments"),
         args.optional("labels"),
-        args.optional("sentry-dsn"),
+        args.asMap.get("sentry-dsn").map(_.mkString(",")),
```
Hacky, but works! Maybe worth leaving a comment that we had to do it in order to support tags?

chuwy reviewed (approved):
Looks great!

oguzhanunlu commented:
Thanks @benjben !

benjben commented:
Hi @chuwy @oguzhanunlu , I added the implementation of #388.
While I was at it, I refactored/cleaned configuration parsing in Beam Enrich, I got rid of `ParsedEnrichConfig` in favor of validating the configuration directly when creating `EnrichConfig`. I also add unit tests to make sure that the app crashes if a config is not valid.

chuwy commented on modules/beam/src/main/scala/com.snowplowanalytics.snowplow.enrich.beam/config.scala:
```diff
@@ -197,4 +200,15 @@ object config {
+    maybeDSN match {
+      case Some(uri) =>
+        Either
+          .catchNonFatal(URI.create(uri))
```
It in fact fails on very few sets invalid URIs, unfortunately, like `/` is valid. I think eventually we want to make this check more restrictive.

benjben replied:
Indeed we could. Should I at least check that it starts with `http://` or `https://` ?

chuwy replied:
I don't know. Probably. If we're sure Sentry supports HTTP(S) only.

benjben replied:
Seems that this is a REST API : https://docs.sentry.io/api/requests/
I'm adding this additional check

chuwy commented on modules/beam/src/test/resources/enrichments_wrong/foo.json:
```diff
@@ -0,0 +1 @@
+bar
```
Don't think we need it?

benjben replied:
It's used here : https://github.com/snowplow/enrich/pull/392/files#diff-5e7b8cfab5c73bb6092007871ff982d471bdb1b42ac80a82350ab64669d87715R95
It's not directly related to this issue but I took the opportunity to improve unit testing to our configuration parsing

chuwy replied:
Ah ok, sorry - I somehow thouoght it's in the repo's root.

chuwy reviewed (approved):
Looks great, just:

1. Leftover file - please delete iit
2. My comment in #388

oguzhanunlu reviewed (approved):
Thanks @benjben !
